cachito/workers/scm.py

```diff
@@ -286,15 +286,25 @@ def update_git_submodules(self, repo):
         `go-github` submodule, the dir structure would look like,
         retrodep/go-github/<content_of_go-github_repo>
 
+        The git command is called directly because of an issue with GitPython's handling of
+        submodules: https://github.com/gitpython-developers/GitPython/issues/1058
+
         :param git.Repo repo: the repository object.
-        :raises RepositoryAccessError: if updating the git submodules fail.
+        :raises SubprocessCallError: if the 'git submodule update --init' command fails to
+                                     update the git submodules.
         """
+        log.debug(f"Git submodules for the requested repo are: {repo.submodules}")
+        cmd = ["git", "submodule", "update", "--init"]
         try:
-            log.debug(f"Git submodules for the requested repo are: {repo.submodules}")
-            repo.submodule_update(recursive=False)
-        except Exception as e:
-            log.exception("Updating the Git submodule(s) from '%s' failed %s", self.url, e)
-            raise RepositoryAccessError("Updating the Git submodule(s) failed")
+            run_cmd(cmd, {"cwd": repo.working_dir, "check": True})
+        except subprocess.CalledProcessError:
+            log.exception(
+                (
+                    f"Updating the Git submodule(s) from {self.url} failed. Cachito attempted "
+                    f"to execute the \"{' '.join(cmd)}\" command but encountered an error."
+                )
+            )
+            raise SubprocessCallError("Updating the Git submodule(s) failed")
```

Comment:
I think run_cmd already wraps CalledProcessErorr in something custom

Reply:
I thought that when check=True that CalledProcessError would be raised for non-zero exit codes :thinking:

Reply:
Oh, you're right, I missed the check=True param

But should we use it? I would almost consider that a misuse of the run_cmd function, since it's normally supposed to raise this for failing commands https://github.com/containerbuildsystem/cachito/blob/1221b0849785165202d8c38ce3558933b5b28a65/cachito/workers/__init__.py#L41-L45

Reply:
Yeah, I agree that it's a bit odd. IIRC the reason that I took that approach was because I saw that run_cmd was being used in a similar way in the same file where git fsck was being called. Looking back through the history, it looked like it was done that way to take advantage of the [command timeout setting](https://github.com/containerbuildsystem/cachito/pull/457) https://github.com/containerbuildsystem/cachito/blob/f189008d440b7c001170d911da5ff4c5fd618a61/cachito/workers/scm.py#L107-L112

Reply:
Ah, fair enough, I didn't realize it was already being used this way. Could we just make sure to log the stderr? I think the CalledProcessError exception does not include the stderr by default, we would have to log it explicitly like with the git fsck call

Reply:
Added :+1: